include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include <bringauto-fleet-http-client-generated/api/CarApi.h>
+#include <bringauto-fleet-http-client-generated/api/DeviceApi.h>
+#include <bringauto-fleet-http-client-generated/api/ModuleApi.h>
+#include <bringauto/fleet_protocol/http_client/RequestFrequencyGuard.hpp>
+
+#include <optional>
+
+
+namespace bringauto::fleet_protocol::http_client {
+
+
+class FleetApiClient {
+public:
+	/**
+     * @brief Configuration struct for the FleetApiClient
+     */
+	struct FleetApiClientConfig {
+		/// URL of Fleet v2 HTTP API
+		std::string apiUrl;
+		/// api key used for authorization
+		std::string apiKey;
+		/// company identifier in the API
+		std::string companyName;
+		/// car identifier in the API
+		std::string carName;
+	};
+
+	/**
+     * @brief Constructs the fleet API client
+     * @param facConfig struct containing the configuration for the fleet api client
+     * @param rfgConfig struct containing the configuration for the request frequency guard (used in getCommands/getStatuses)
+     */
```

Comment:
there is still formatting issue, spaces are used instead of tabs

Reply:
tabs should now be everywhere